File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/db/WalletDao.java
package at.msd.friehs_bicha.cdcsvparser.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;

import java.util.List;

import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet;
import at.msd.friehs_bicha.cdcsvparser.wallet.WalletWithTransactions;

@Dao
public interface WalletDao {

    @Query("SELECT * FROM wallet")
    List<WalletWithTransactions> getAll();

    @Query("SELECT * FROM wallet WHERE walletId IN (:walletIds)")
    List<WalletWithTransactions> loadAllByIds(int[] walletIds);

    @Transaction
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<Wallet> wallets);


    @Delete
    void delete(Wallet wallet);

    @Query("DELETE FROM wallet")
    void deleteAll();

    @Transaction
    @Query("SELECT * FROM wallet")
    LiveData<List<WalletWithTransactions>> getWalletsWithTransactions();

    @Query("SELECT * FROM wallet")
    LiveData<List<Wallet>> getAllWallets();

}

File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/db/WalletDao.kt
package at.msd.friehs_bicha.cdcsvparser.db

import androidx.lifecycle.LiveData
import androidx.room.*
import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet

@Dao
interface WalletDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertWallet(wallet: Wallet)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(wallets: List<Wallet>)

    @Transaction
    @Query("SELECT * FROM wallets")
    fun getAllWallets(): List<WalletWithTransactions>

    @Transaction
    @Query("SELECT * FROM wallets")
    fun getAllWalletsWithTransactions(): LiveData<List<WalletWithTransactions>>

    @Query("DELETE FROM wallets")
    fun deleteAll()

}